src/compiler/checker.ts

```diff
@@ -43474,6 +43481,41 @@ namespace ts {
             return resolveName(location, reference.escapedText, SymbolFlags.Value | SymbolFlags.ExportValue | SymbolFlags.Alias, /*nodeNotFoundMessage*/ undefined, /*nameArg*/ undefined, /*isUse*/ true);
         }
 
+        /**
+         * Get either a value-meaning symbol or an alias symbol.
+         * Unlike `getReferencedValueSymbol`, if the cached resolved symbol is the unknown symbol,
+         * we call `resolveName` to find a symbol.
+         * This is because when caching the resolved symbol, we only consider value symbols, but here
+         * we want to also get an alias symbol if one exists.
+         */
+        function getReferencedSymbol(reference: Identifier, startInDeclarationContainer?: boolean): Symbol | undefined {

```

Comment:
We need to call `resolveName` here when the resolved symbol is unknown (i.e. there's a cache miss), because the resolved symbol is cached by doing a call to `resolveName` that only looks for symbols that have a value meaning.
Since we stopped eliding imports in JS files, that means some of the import symbols we want to find won't have a value meaning, but we still want to find them anyways, e.g. imports of types. Since those are imports, they'll have an alias meaning, hence we call `resolveName` with value or alias meaning.

Reply:
Could we know during cache construction whether we'll eventually want alias symbols?  Would that interfere with other cache consumers?

Reply:
For (1), I think maybe that could be done. We want aliases for every identifier that goes through a module transform, worst case we need it for every identifier that survived the type erasure transform. But for (2) I think we would need a different cache, because changing it to cache alias-meaning symbols would interfere with the way the checker uses this cache.

Reply:
Probably not worth the extra complexity of maintaining another cache (conditional on no-emit?), especially before we measure a problem.